### PR TITLE
Added support to select database roles from `tsh`.

### LINKED ISSUE #35566: Ability to select database roles for auto-user-provisioning database session
What would you like Teleport to do?
When using auto-user provisioning feature, able to select a subset of allowed database roles for a database session, e.g.:
```bash
$ tsh db connect --db-name dev --db-role roleA -r roleB my-postgres
```

What problem does this solve?

A Teleport user may be granted multiple database roles for a select database. When Database Service provisions the user, it tries to assign all allowed database roles and will fail the session if any of the roles do not exist on the database.

If a workaround exists, please include it.

Teleport roles can be carefully crafted so that they only match the databases that have these roles preset in the database. Not feasible when no common database roles exist on a large number of databases.
```[tasklist]
### Tasks
- [x] Show database roles in `tsh db ls -v`
- [x] `tsh db connect --db-role`
- [ ] Teleport Connect (now tracked separately #36703)
```

### PR DESCRIPTION
**Pull request by greedy52**
Related
- #35566 
- #35582

changelog: Added support to select database roles from `tsh`.

Example:
Default `db_roles` is ["role1", "role3"]
```bash
$ tsh db connect --db-user STeve --db-name test aurora-postgres-auto --db-roles role1 
psql (16.1, server 15.2)
SSL connection (protocol: TLSv1.3, cipher: TLS_AES_128_GCM_SHA256, compression: off)
Type "help" for help.

test=> select rolname from pg_auth_members join pg_roles on roleid = oid where member=(select oid from pg_roles where rolname = (select current_user));
      rolname       
--------------------
 rds_iam
 role1
 teleport-auto-user
(3 rows)

test=> ^D\q

$ tsh proxy db --db-roles role3 aurora-postgres-auto
...
```

Audit:
Screenshot 2023-12-18 at 4 04 00 PM

### CODE REVIEW

File: tool/tsh/common/db_test.go [hunk 1]
```diff
@@ -106,6 +107,13 @@ func testDatabaseLogin(t *testing.T) {
 					Name:     "mysql",
 					Protocol: defaults.ProtocolMySQL,
 					URI:      "localhost:3306",
+				}, {
+					Name:     "mysql-autouser",
```
Comment (greedy52) [inline tool/tsh/common/db_test.go:111]: oops, seems I forgot to add actual test case for this. Will add.

File: tool/tsh/common/tsh.go [hunk 2]
```diff
@@ -3015,15 +3020,21 @@ func printDatabasesWithClusters(clusterFlag string, dbListings []databaseListing
...
-	case active.Database != "":
-		return fmt.Sprintf("> %v (db: %v)", active.ServiceName, active.Database)
-	}
-	return fmt.Sprintf("> %v", active.ServiceName)
+	var details []string
+	if active.Username != "" {
+		details = append(details, "user: "+active.Username)
+	}
+	if active.Database != "" {
+		details = append(details, "db: "+active.Database)
```
Comment (smallinsky) [inline tool/tsh/common/tsh.go:3028]: nit:
 just inconsistency with below style: 
` fmt.Sprintf("db: %v", +active.Database)` vs `"db ": "+active.Database"`

File: tool/tsh/common/db.go [hunk 3]
```diff
@@ -1286,9 +1298,25 @@ func dbInfoHasChanged(cf *CLIConf, certPath string) (bool, error) {
...
+	if len(s1) != len(s2) {
+		return false
+	}
+	for i := range s1 {
+		if !slices.Contains(s2, s1[i]) {
+			return false
+		}
+	}
+	return true
+}
```
Comment (smallinsky) [inline tool/tsh/common/db.go:1318]: nit: Can we move this to utils package ?

**Review by smallinsky — APPROVED** [review]

**Comment by greedy52** [comment]
Friendly ping @gabrielcorado

**Review by r0mant — COMMENTED** [review]
Overall lgtm but one comment about the CLI flag.

File: tool/tsh/common/tsh.go [hunk 4]
```diff
@@ -817,8 +819,9 @@ func Run(ctx context.Context, args []string, opts ...CliOption) error {
 	proxyDB.Arg("db", "The name of the database to start local proxy for").StringVar(&cf.DatabaseService)
 	proxyDB.Flag("port", "Specifies the source port used by proxy db listener").Short('p').StringVar(&cf.LocalProxyPort)
 	proxyDB.Flag("tunnel", "Open authenticated tunnel using database's client certificate so clients don't need to authenticate").BoolVar(&cf.LocalProxyTunnel)
-	proxyDB.Flag("db-user", "Database user to log in as.").StringVar(&cf.DatabaseUser)
-	proxyDB.Flag("db-name", "Database name to log in to.").StringVar(&cf.DatabaseName)
+	proxyDB.Flag("db-user", "Database user to log in as.").Short('u').StringVar(&cf.DatabaseUser)
+	proxyDB.Flag("db-name", "Database name to log in to.").Short('n').StringVar(&cf.DatabaseName)
+	proxyDB.Flag("db-role", "List of database roles to use for auto-provisioned user (e.g. --db-role roleA -r roleB).").Short('r').StringsVar(&cf.DatabaseRoles)
```
[the remark this answers is not in the document]
Comment (greedy52) [reply tool/tsh/common/tsh.go:824]: Ya, I was struggling with which way to go. I decided on multiple flags as "comma" can be a valid character in a role name (e.g `role,1`). Though I doubt ppl actually use that in practice. I think it also depends on whether we expect the user to pick a single role at a time or multiple ones.  I can certainly change it to `--db-roles role1,role2`. What do you think?

[the remark this answers is not in the document]
Comment (r0mant) [reply tool/tsh/common/tsh.go:824]: To be honest I think comma in a role name is such an edge-case that I would still prefer a slightly better UX with `--db-roles roleA,roleB`.

[the remark this answers is not in the document]
Comment (GavinFrazar) [reply tool/tsh/common/tsh.go:824]: does kingpin support `--db-role A --db-role B`?

[the remark this answers is not in the document]
Comment (greedy52) [reply tool/tsh/common/tsh.go]: Updated https://github.com/gravitational/teleport/pull/35867/commits/9f3ef1c3199c7f2e2483dd719944390e1ecf3ad3. 

> does kingpin support --db-role A --db-role B?

Yes. I thought that's the native kingpin way for `StringsVar`. I think this is another good candidate for CLI guideline. Currently `tsh` is not consistent on passing a list of strings.

**Review by r0mant — APPROVED** [review]

**Pull request merged**